README.md

```diff
@@ -16,20 +16,121 @@ If no patterns are specified, the default pattern of `^(fmt\.Print.*|print|print
 functions (and whole files), that are identifies as Godoc examples (https://blog.golang.org/examples) are excluded from 
 checking.
 
-A larger set of interesting patterns might include:
+By default, patterns get matched against the actual expression as it appears in
+the source code. The effect is that ``^fmt\.Print.*$` will not match when that
+package gets imported with `import fmt2 "fmt"` and then the function gets
+called with `fmt2.Print`.
+
+This makes it hard to match packages that may get imported under a variety of
+different names, for example because there is no established convention or the
+name is so generic that import aliases have to be used. To solve this,
+forbidigo also supports a more advanced mode where it uses type information to
+identify what an expression references. This needs to be enabled through the
+`expand_expression` command line parameter. Beware this may have a performance
+impact because additional information is required for the analysis.
+
+Replacing the literal source code works for items in a package as in the
+`fmt2.Print` example above and also for struct fields and methods. For those,
+`<package>.<type name>.<field or method name>` replaces the source code
```

Comment:
Thanks for noting the similarity to godoc for this.

Reply:
Well, this here isn't anymore what godoc uses - it's the name from the package, where godoc uses the import path.

Both approaches work for me, but as you previously stated that you want patterns to match the package name and I found out how to implement that, I went with that.

Reply:
Oh darn! I didn't actually push that change. I've pushed it now.